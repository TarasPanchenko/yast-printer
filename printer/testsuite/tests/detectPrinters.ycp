/**
 * File:
 *  detectPrinters.ycp
 *
 * Module:
 *  Printer configurator
 *
 * Summary:
 *  Miscelaneous tests.
 *
 * Authors:
 *  Petr Blahos <pblahos@suse.cz>
 *
 * $Id$
 *
 */

{
    import "Pkg";

    include "testsuite.ycp";

    // testedfiles: Printer.ycp Spooler.ycp Printerlib.ycp CUPS.ycp

    map I_READ = $[
	"target" : $[
	    "size" : -1,
	    "tmpdir" : "/tmp/",
	    "yast2" : nil,
	    "string" : "",
	],
	"sysconfig" : $[
	    "console" : $[
		"CONSOLE_FONT" : "",
		"CONSOLE_SCREENMAP" : "",
		"CONSOLE_UNICODEMAP" : "",
		"CONSOLE_MAGIC" : "",
		"CONSOLE_ENCODING" : "",
	    ],
	    "language" : $[
		"RC_LANG" : "",
		"DEFAULT_LANGUAGE" : "",
	    ],
	    "personal-firewall" : $[
		"REJECT_ALL_INCOMING_CONNECTIONS" : nil,
	    ],
	],
	"etc" : $[
	    "install_inf" : $[
		"Cmdline" : "",
		"InstMode" : nil,
	    ],
	],
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	    "is_uml" : false,
	],
	"product" : $[
	    "features" : $[
		"USE_DESKTOP_SCHEDULER" : "0",
		"ENABLE_AUTOLOGIN" : "0",
		"EVMS_CONFIG" : "0",
		"IO_SCHEDULER" : "cfg",
		"UI_MODE" : "expert",
	    ],
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[
        "target" : $[
            "bash_output" : $[],
        ],
    ];
    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], 0);
    import "Printer";
    import "Printerlib";
    import "Spooler";

    map READ =
        $[
            "probe" : $[
                "printer" : [
                    $[
                        "bus":"Parallel",
                        "class_id":265,
                        "dev_name":"/dev/lp0",
                        "device":"BJC-6100",
                        "sub_class_id":0,
                        "unique_key":"xVOt.CRhzf0_Mb88",
                        "vendor":"Canon"
                    ],
                    $[
                        "bus":"USB",
                        "class_id":265,
                        "dev_name":"/dev/usblp0",
                        "device_id":200786,
                        "resource":$["baud":[$["speed":1500000]]],
                        "rev":"1.03",
                        "sub_class_id":0,
                        "sub_device":"BJC-6100",
                        "sub_vendor":"Canon",
                        "unique_key":"Q0jS.ZoQ4aZ8KEv3",
                        "vendor":"Canon Inc.",
                        "vendor_id":197801
                    ]
                ]
            ],
	    "proc" : $[ "modules" : $[ "usblp" : ""]],
	    "ppd" : $[
		"db" : $[
		    "vendorname" : "",
		    "modelname" : "",
		],
	    ],
	    "target" : $[
		"size" : -1,
		"string" : "",
	    ],
	];
    map EXEC= $[
	"target" : $[
	    "bash_output" : $[
		"exit" : 0,
		"stdout" : "",
	    ],
	],
    ];
    DUMP ("--- Detection for lprng");
    Spooler::Set ("lprng");
    TEST (``(Printer::Detect ()), [READ, $[], EXEC], true);
    DUMP ("--- Detection for cups");
    Spooler::Set ("cups");
    TEST (``(Printer::Detect ()), [READ, $[], EXEC], true);

    READ = $[
            "probe" : $[
                "printer" : [
                    $[
                        "bus":"USB",
                        "class_id":265,
                        "dev_name":"/dev/usblp0",
                        "device_id":200786,
                        "resource":$["baud":[$["speed":1500000]]],
                        "rev":"1.03",
                        "sub_class_id":0,
                        "sub_device":"USB Printer",
                        "sub_vendor":"Epson",
                        "unique_key":"Q0jS.ZoQ4aZ8KEv3",
                        "vendor":"Epson Inc.",
                        "vendor_id":197801
                    ],
                ],
	    ],
	    "proc" : $[ "modules" : $[ "usblp" : ""]],
    	    "ppd" : $[
		"db" : $[
		    "vendorname" : "",
		    "modelname" : "",
		],
	    ],
	    "target" : $[ "size" : -1],
        ];
    DUMP ("--- Detection for lprng");
    Spooler::Set ("lprng");
    TEST (``(Printer::Detect ()), [READ, $[], EXEC], true);
    DUMP ("--- Detection for cups");
    Spooler::Set ("cups");
    TEST (``(Printer::Detect ()), [READ, $[], EXEC], true);

    // try to detect for cups but without vendor_id and device_id
    READ = $[
            "probe" : $[
                "printer" : [
                    $[
                        "bus":"USB",
                        "class_id":265,
                        "dev_name":"/dev/usblp0",
                        "resource":$["baud":[$["speed":1500000]]],
                        "rev":"1.03",
                        "sub_class_id":0,
                        "sub_device":"USB Printer",
                        "sub_vendor":"Epson",
                        "unique_key":"Q0jS.ZoQ4aZ8KEv3",
                        "vendor":"Epson Inc.",
                    ]
                ]
            ],
	    "proc" : $[ "modules" : $[ "usblp" : ""]],
	    "target" : $[ "size" : 555],
       	    "ppd" : $[
		"db" : $[
		    "vendorname" : "",
		    "modelname" : "",
		],
	    ],
     ];
    EXEC = $[
	"target" : $[
	    "bash_output" : $[
		"exit" : 0,
		"stderr" : "",
		"stdout" : "direct usb://EPSON/Stylus%20Photo%20810?serial=W33040110201335050 \"EPSON Stylus Photo 810\" \"USB Printer #1\"
direct usb:/dev/usb/lp1 \"Unknown\" \"USB Printer #2\"
direct usb:/dev/usb/lp2 \"Unknown\" \"USB Printer #3\"",
		]
	    ],
	];
    DUMP ("--- Detection for cups");
    Spooler::Set ("cups");
    TEST (``(Printer::Detect ()), [READ, $[], EXEC], true);

    // try to detect all-in-one device
    READ = $[
            "probe" : $[
                "printer" : [
                    $[
                        "bus":"USB",
                        "class_id":265,
                        "dev_name":"/dev/usblp0",
                        "resource":$["baud":[$["speed":1500000]]],
                        "rev":"1.03",
                        "sub_class_id":0,
                        "sub_device":"USB Printer",
                        "sub_vendor":"HP",
                        "unique_key":"Q0jS.ZoQ4aZ8KEv3",
                        "vendor":"HP",
			"model" : "Hewlett-Packard PSC 2200 Series",
                    ]
                ]
            ],
            "proc" : $[ "modules" : $[ "usblp" : ""]],
            "target" : $[ "size" : 555],
       	    "ppd" : $[
		"db" : $[
		    "vendorname" : "",
		    "modelname" : "",
		],
	    ],
     ];

    EXEC = $[
	"target" : $[
	    "bash_output" : $[
		"exit" : 0,
		"stderr" : "",
		"stdout" : "direct ptal:/mlc:usb:PSC_2200_Series \"Hewlett-Packard PSC 2200 Series\" \"PTAL mlc:usb:PSC_2200_Series\"
"
	    ],
	]
    ];

    DUMP ("--- Detection for cups");
    Spooler::Set ("cups");
    Printerlib::hp_all_in_one = ["USB Printer"];
    TEST (``(Printer::Detect ()), [READ, $[], EXEC], true);


}
