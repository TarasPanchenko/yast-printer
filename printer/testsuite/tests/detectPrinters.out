Dump	--- Detection for lprng
Execute	.target.bash "test -f /etc/init.d/hotplug && (test -d /proc/bus/usb/001 || /etc/init.d/hotplug start)" 0
Read	.probe.printer [$["bus":"Parallel", "class_id":265, "dev_name":"/dev/lp0", "device":"BJC-6100", "sub_class_id":0, "unique_key":"xVOt.CRhzf0_Mb88", "vendor":"Canon"], $["bus":"USB", "class_id":265, "dev_name":"/dev/usblp0", "device_id":200786, "resource":$["baud":[$["speed":1500000]]], "rev":"1.03", "sub_class_id":0, "sub_device":"BJC-6100", "sub_vendor":"Canon", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "vendor":"Canon Inc.", "vendor_id":197801]]
Read	.target.size "/usr/lib/cups/backend/usb" -1
Read	.ppd.db.vendorname "Canon" ""
Read	.ppd.db.modelname ["", "BJC-6100"] ""
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Read	.ppd.db.vendorname "Canon" ""
Read	.ppd.db.modelname ["", "BJC-6100"] ""
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Return	[$["device":"BJC-6100", "device_db":"", "info":"Canon BJC-6100", "location":"Parallel printer on /dev/lp0", "unique_key":"xVOt.CRhzf0_Mb88", "uri":"parallel:/dev/lp0", "vendor":"Canon", "vendor_db":""], $["device":"BJC-6100", "device_db":"", "info":"Canon BJC-6100", "location":"USB printer on /dev/usblp0", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "uri":"usb:/dev/usblp0", "vendor":"Canon", "vendor_db":""]]
Dump	--- Detection for cups
Execute	.target.bash "test -f /etc/init.d/hotplug && (test -d /proc/bus/usb/001 || /etc/init.d/hotplug start)" 0
Read	.probe.printer [$["bus":"Parallel", "class_id":265, "dev_name":"/dev/lp0", "device":"BJC-6100", "sub_class_id":0, "unique_key":"xVOt.CRhzf0_Mb88", "vendor":"Canon"], $["bus":"USB", "class_id":265, "dev_name":"/dev/usblp0", "device_id":200786, "resource":$["baud":[$["speed":1500000]]], "rev":"1.03", "sub_class_id":0, "sub_device":"BJC-6100", "sub_vendor":"Canon", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "vendor":"Canon Inc.", "vendor_id":197801]]
Read	.target.size "/usr/lib/cups/backend/usb" -1
Read	.ppd.db.vendorname "Canon" ""
Read	.ppd.db.modelname ["", "BJC-6100"] ""
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Read	.ppd.db.vendorname "Canon" ""
Read	.ppd.db.modelname ["", "BJC-6100"] ""
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Return	[$["device":"BJC-6100", "device_db":"", "info":"Canon BJC-6100", "location":"Parallel printer on /dev/lp0", "unique_key":"xVOt.CRhzf0_Mb88", "uri":"parallel:/dev/lp0", "vendor":"Canon", "vendor_db":""], $["device":"BJC-6100", "device_db":"", "info":"Canon BJC-6100", "location":"USB printer on /dev/usblp0", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "uri":"usb:/dev/usblp0", "vendor":"Canon", "vendor_db":""]]
Dump	--- Detection for lprng
Execute	.target.bash "test -f /etc/init.d/hotplug && (test -d /proc/bus/usb/001 || /etc/init.d/hotplug start)" 0
Read	.probe.printer [$["bus":"USB", "class_id":265, "dev_name":"/dev/usblp0", "device_id":200786, "resource":$["baud":[$["speed":1500000]]], "rev":"1.03", "sub_class_id":0, "sub_device":"USB Printer", "sub_vendor":"Epson", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "vendor":"Epson Inc.", "vendor_id":197801]]
Read	.target.size "/usr/lib/cups/backend/usb" -1
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Read	.proc.modules $["usblp":""]
Execute	.target.bash "\nMAIN_PID=$$\n{/bin/echo -en \"\\000\\000\\000\\033\\001\\100\\105\\112\\114\\040\\061\\062\\070\\064\\056\\064\\012\\100\\105\\112\\114\\040\\040\\040\\040\\040\\012\" > /dev/usblp0 ; kill MAIN_PID} &\nECHO_PID=$!\n{sleep 2s ; kill $ECHO_PID &>/dev/null ; } &" 0
Read	.ppd.db.vendorname "Epson" ""
Read	.ppd.db.modelname ["", "USB Printer"] ""
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Return	[$["device":"USB Printer", "device_db":"", "info":"Epson USB Printer", "location":"USB printer on /dev/usblp0", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "uri":"usb:/dev/usblp0", "vendor":"Epson", "vendor_db":""]]
Dump	--- Detection for cups
Execute	.target.bash "test -f /etc/init.d/hotplug && (test -d /proc/bus/usb/001 || /etc/init.d/hotplug start)" 0
Read	.probe.printer [$["bus":"USB", "class_id":265, "dev_name":"/dev/usblp0", "device_id":200786, "resource":$["baud":[$["speed":1500000]]], "rev":"1.03", "sub_class_id":0, "sub_device":"USB Printer", "sub_vendor":"Epson", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "vendor":"Epson Inc.", "vendor_id":197801]]
Read	.target.size "/usr/lib/cups/backend/usb" -1
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Read	.proc.modules $["usblp":""]
Execute	.target.bash "\nMAIN_PID=$$\n{/bin/echo -en \"\\000\\000\\000\\033\\001\\100\\105\\112\\114\\040\\061\\062\\070\\064\\056\\064\\012\\100\\105\\112\\114\\040\\040\\040\\040\\040\\012\" > /dev/usblp0 ; kill MAIN_PID} &\nECHO_PID=$!\n{sleep 2s ; kill $ECHO_PID &>/dev/null ; } &" 0
Read	.ppd.db.vendorname "Epson" ""
Read	.ppd.db.modelname ["", "USB Printer"] ""
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Return	[$["device":"USB Printer", "device_db":"", "info":"Epson USB Printer", "location":"USB printer on /dev/usblp0", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "uri":"usb:/dev/usblp0", "vendor":"Epson", "vendor_db":""]]
Dump	--- Detection for cups
Execute	.target.bash "test -f /etc/init.d/hotplug && (test -d /proc/bus/usb/001 || /etc/init.d/hotplug start)" 0
Read	.probe.printer [$["bus":"USB", "class_id":265, "dev_name":"/dev/usblp0", "resource":$["baud":[$["speed":1500000]]], "rev":"1.03", "sub_class_id":0, "sub_device":"USB Printer", "sub_vendor":"Epson", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "vendor":"Epson Inc."]]
Read	.target.size "/usr/lib/cups/backend/usb" 555
Execute	.target.bash_output "/usr/lib/cups/backend/usb" $["exit":0, "stderr":"", "stdout":"direct usb://EPSON/Stylus%20Photo%20810?serial=W33040110201335050 \"EPSON Stylus Photo 810\" \"USB Printer #1\"\ndirect usb:/dev/usb/lp1 \"Unknown\" \"USB Printer #2\"\ndirect usb:/dev/usb/lp2 \"Unknown\" \"USB Printer #3\""]
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Read	.proc.modules $["usblp":""]
Execute	.target.bash "\nMAIN_PID=$$\n{/bin/echo -en \"\\000\\000\\000\\033\\001\\100\\105\\112\\114\\040\\061\\062\\070\\064\\056\\064\\012\\100\\105\\112\\114\\040\\040\\040\\040\\040\\012\" > /dev/usblp0 ; kill MAIN_PID} &\nECHO_PID=$!\n{sleep 2s ; kill $ECHO_PID &>/dev/null ; } &" 0
Read	.ppd.db.vendorname "Epson" ""
Read	.ppd.db.modelname ["", "Stylus Photo 810"] ""
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Return	[$["device":"Stylus Photo 810", "device_db":"", "info":"Epson Stylus Photo 810", "location":"USB printer on //EPSON/Stylus%20Photo%20810?serial=W33040110201335050", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "uri":"usb://EPSON/Stylus%20Photo%20810?serial=W33040110201335050", "vendor":"Epson", "vendor_db":""]]
Dump	--- Detection for cups
Execute	.target.bash "test -f /etc/init.d/hotplug && (test -d /proc/bus/usb/001 || /etc/init.d/hotplug start)" 0
Read	.probe.printer [$["bus":"USB", "class_id":265, "dev_name":"/dev/usblp0", "model":"Hewlett-Packard PSC 2200 Series", "resource":$["baud":[$["speed":1500000]]], "rev":"1.03", "sub_class_id":0, "sub_device":"USB Printer", "sub_vendor":"HP", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "vendor":"HP"]]
Read	.target.size "/usr/lib/cups/backend/usb" 555
Execute	.target.bash_output "/usr/lib/cups/backend/usb" $["exit":0, "stderr":"", "stdout":"direct ptal:/mlc:usb:PSC_2200_Series \"Hewlett-Packard PSC 2200 Series\" \"PTAL mlc:usb:PSC_2200_Series\"\n"]
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Read	.ppd.db.vendorname "HP" ""
Read	.ppd.db.modelname ["", "USB Printer"] ""
Execute	.target.bash "grep '^usblp ' /proc/modules" 0
Return	[$["device":"USB Printer", "device_db":"", "info":"HP USB Printer", "location":"USB printer on /dev/usblp0", "unique_key":"Q0jS.ZoQ4aZ8KEv3", "uri":"usb:/dev/usblp0", "vendor":"HP", "vendor_db":""]]
